partner-chains-cli/src/tests/mod.rs

```diff
@@ -132,14 +132,44 @@ impl MockIOContext {
 			None => None,
 		}
 	}
-	pub fn no_more_io_expected(&self) {
+}
+
+macro_rules! should_have_no_io_left {
+	($context:expr) => {{
 		assert!(
```

Comment:
Yes, assertion is repeated 3 times, but reusing `should_have_no_io_left` requires it to be imported (macros don't close on their scope like functions)

Reply:
I'm not sure what the best option would behere tbh. Open for suggestions

Reply:
I don't like the two following macros, because the don't assert on the `Ok` of `Err`. I see in diff, that we failed to make better assertions. The two following macros help in hiding our omission.
Even in cases where it's enough to assert on `in_ok`, `is_err`, I don't think lines save are worth another layer of indirection.

Reply:
Okay, removed them and replaced them with `expect` and `expect_err`.

Btw. I don't think `is_err` and `is_ok` is ever a good thing to assert, because the first thing I do when assertion like that fails is to replace it with an `expect*` to see what error or what result is returned anyway.